tests/api/runtimes/base.py

```diff
@@ -705,6 +716,16 @@ def _assert_pod_creation_config(
         if expected_priority_class_name:
             assert pod.spec.priority_class_name == expected_priority_class_name
 
+        if expected_security_context:
+            assert (
+                deepdiff.DeepDiff(
+                    pod.spec.security_context.to_dict(),
+                    expected_security_context.to_dict(),
+                    ignore_order=True,
+                )
+                == {}
+            )
+
```

Comment:
I actually have a bit of issue with the `_assert_pod_creation_config` method. We are only asserting if we expect some attribute, which is problematic because it means we are only covering one situation and ignoring the specific mistakes we are attempting to catch in these kind of tests.
This requires some fundamental change in the tests structure so I am not expecting for this change for all other attributes in this PR, but let's start with security_context.
Take a look on what I did for node selection in `assert_node_selection`.